src/storage/v3/shard.cpp

```diff
@@ -1048,6 +1102,21 @@ void Shard::StoreMapping(std::unordered_map<uint64_t, std::string> id_to_name) {
   name_id_mapper_.StoreMapping(std::move(id_to_name));
 }
 
+std::optional<SplitInfo> Shard::ShouldSplit() const noexcept {
+  if (vertices_.size() > config_.split.max_shard_vertex_size) {
+    auto mid_elem = vertices_.begin();
+    std::ranges::advance(mid_elem, static_cast<VertexContainer::difference_type>(vertices_.size() / 2));
+    return SplitInfo{mid_elem->first, shard_version_};
+  }
+  return std::nullopt;
+}
```

Comment:
in the near future, we will want to only calculate the split point once, because this is run every time `Cron` runs, and if we iterate over half of the entire index every ~200ish milliseconds it will kill the shard before the split is completed, because it will take the overall cluster a few round trips to accomplish that. I wouldn't fix that now but I'll add a TODO code suggestion related to it.